tests/jcond.s

```diff
@@ -0,0 +1,31 @@
+jmp #5                      ; keep pass/fail at fixed location
+print ^cond_jump_insts_ok   ; overall pass
+halt
+print ^cond_jump_insts_fail ; any fail
+halt
+sub $1 $2 #3                ; test all j insts that don't consider < case as match
```

Comment:
"<" stands for "op1 < op2", which is the condition that "sub $1, $2, #3" creates.

When op1 ($1) is smaller than op2 ($2), the condition does not match the jump criteria that jz, jge, or jgt needs. Hence, these three instruction won't branch to the 'fail' case.

If they branches (which shouldn't happen), it means they were implemented wrongly.

The same logic applies to other test cases.